server/game/gameSystems/DamageSystem.ts

```diff
@@ -149,6 +152,10 @@ export class DamageSystem<TContext extends AbilityContext = AbilityContext, TPro
             default:
                 Contract.fail(`Unexpected damage type: ${properties['type']}`);
         }
+
+        // Check if the damage will defeat the card, this can be used by abilities (e.g. Tarfful) to determine if the card will be defeated or not
+        const damageAmount = event.amount ?? event.sourceEventForExcessDamage?.availableExcessDamage ?? 0;
```

Comment:
Would `this.getDamageAmountFromEvent()` work? That also has contract checks to confirm one or the other of these values exists

Reply:
I haven't tried but I don't think so because `getDamageAmountFromEvent` modifies the event to remove available excess damage so the event handler wouldn't work correctly.

Reply:
Gotcha, good catch. I'm pretty sure the 0 case should never happen here, so I definitely want to at least redo the contract checks from that method. So we could either copy them here or maybe add a parameter to the method to disable removing the excess damage. Or maybe split the latter into a separate method, that might make the most sense.

Reply:
I moved the part the removes the excess damage to the event handler, I think it makes more sense to have it there.